.github/workflows/_deploy_kafka_test.yml

```diff
@@ -0,0 +1,134 @@
+        type: string
+      sidecar_registry:
+        required: true
+        default: "mcr.microsoft.com"
+        description: "The registry to get the image from"
+        type: string
+      key-release-image: 
+        required: true 
+        description: "The image of the SKR sidecar to use"
+        type: string
+      consumer-image: 
+        description: "Consumer Image"
+        required: true
+        type: string
+      producer-image: 
+        description: "Producer Image"
+        required: true
+        type: string
+jobs:
+  deploy-kafka:
+    name: Deploy Kafka Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: main
+
+      - name: Log into Azure
+        run: |
+          az login --service-principal \
+            --username ${{ secrets.AZURE_SERVICE_PRINCIPAL_APP_ID }} \
+            --password ${{ secrets.AZURE_SERVICE_PRINCIPAL_PASSWORD }} \
+            --tenant ${{ secrets.AZURE_SERVICE_PRINCIPAL_TENANT }}
+            
+      - name: Install Dependencies 
+        id: install-dependencies 
+        run: |
+            curl -L https://github.com/a8m/envsubst/releases/download/v1.2.0/envsubst-`uname -s`-`uname -m` -o envsubst
+            chmod +x envsubst
+            sudo mv envsubst /usr/local/bin
+
+            result=$(az extension list -o table  2>&1 || true)
+            if [[ $result == *"confcom"* ]]; then
+              echo "confcom already installed, upgrading confcom version."
+              az extension update --name confcom
+            else
+              echo "confcom extension not found. Installing confcom..."
+              az extension add --name confcom
+            fi
+
+            curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+            chmod +x kubectl
+            sudo mv kubectl /usr/local/bin
+            kubectl version 2>&1 || true
+
+      - name: Run Workload
+        env:
+          AZURE_AKV_RESOURCE_ENDPOINT: ${{ secrets.SKR_CLIENT_AKV_ENDPOINT }}
+          MAA_ENDPOINT: ${{ secrets.SKR_CLIENT_MAA_ENDPOINT }}
+          CLUSTER_NAME: ${{ inputs.cluster-name }}
+          RESOURCE_GROUP: ${{ vars.RESOURCE_GROUP }} 
+          SIDECAR_IMAGE: ${{ format('{0}/{1}', inputs.sidecar-registry, inputs.key-release-image) }}
+          CONSUMER_IMAGE: ${{ format('{0}/{1}', inputs.sidecar-registry, inputs.consumer-image) }}
+          PRODUCER_IMAGE: ${{ format('{0}/{1}', inputs.sidecar-registry, inputs.producer-image) }}
+        id: run-workload
+        run: | 
+          az aks get-credentials --name $CLUSTER_NAME --resource-group $RESOURCE_GROUP  --overwrite-existing
+          cd main/kafka
+          rm consumer-example.yaml 2>&1 || true 
+          envsubst -i consumer/consumer-pipeline.yaml -o consumer-example.yaml
+          echo "Generating Security Policy for consumer"
```

Comment:
I would make the policy generation a separate step from the "run" step.

Reply:
It's put into the script like you suggested below